src/app/zap-templates/templates/app/gen_config.zapt

```diff
@@ -50,7 +50,24 @@
 {{else if (isStrEqual name "Scenes")}}
 {{#if (isServer side)}}
 // User options for {{side}} plugin {{name}}
-#define EMBER_AF_PLUGIN_SCENES_TABLE_SIZE 3
+// Cluster spec 1.4.8.2
+#ifdef CHIP_CONFIG_MAX_SCENES_PER_FABRIC
+#define MATTER_SCENES_TABLE_SIZE CHIP_CONFIG_MAX_SCENES_PER_FABRIC
+#else
+#define MATTER_SCENES_TABLE_SIZE 16
+#endif
+{{/if}}
+{{#all_user_cluster_generated_attributes}}
+{{#if (isStrEqual ../name clusterName)}}
+{{#if (isStrEqual name "FeatureMap")}}
```

Comment:
This isn't a very scalable option unfortunately, since it requires every cluster that wants to do static pre-processor stuff off of features being enabled to do custom work here.

Would it be possible for you to spend a little bit of extra cycles to emit a bunch of defines for each feature that is enabled based on the ZAP file?

i.e <CLUSTER_NAME>_<FEATURE_NAME>_IS_ENABLED

```
#define SCENES_CLUSTER_NAME_SUPPORT_FEATURE_IS_ENABLED 1
```

That would go a long way towards a scalable solution.

Reply:
Yeah, problem here is that the attribute value toggling the feature is a bit map so you need to check with a specific mask that could change accross clusters. With cluster having the featureMap attributes enabling one feature when others could be enablings multiples features. Moreover with solely the FeatureMap attribute there is no way or knowing what feature is enable by it unless you verify with the specs. So I would say that by default the FeatureMap attribute isn't quite scalable. 

Unless we somehow integrate within the XML definition of the cluster the corresponding bit mapping of the FeatureMap attribute with a string name of the feature that it enables. But doing so should be in another PR and subject to the DM team approval.

Reply:
@mrjerryjohns What could also be done would be to generate a define like you suggested inside the config file 

`#define SCENES_CLUSTER_FEATURE_MAP <Value from ZAP>`


and inside the scene.h file have the appropriate macro for it 

`#define SCENES_CLUSTER_NAME_SUPPORT_FEATURE_IS_ENABLED  (SCENES_CLUSTER_FEATURE_MAP & SPEC_DEFINED_MASK)`

This seems like a more scalable solution that could be applied to all clusters quite easily

Reply:
Will apply the changes in a follow up PR

Reply:
Yup, that works too for me.